tools/visual-pipeline-and-platform-evaluation-tool/collector/qmassa_reader.py

```diff
@@ -103,7 +107,7 @@ def save_last_state(index, timestamp):
                     for key, val in power2[-1].items():
                         print(f"power,type={key},host={HOSTNAME},gpu_id=0 value={val} {ts}")
             else:
-                 # Use the last device state
+                # Use the last device state
```

Comment:
A lot of code in the `if` and `else` blocks is repeated - maybe it could be simplified?
Also, for `devs_state[-2]` in the `if`, I don’t really see the point of printing the power, since it will always be 0 in the first iteration, which just adds confusion. 
But I’d leave a comment explaining why we're not printing the power there.